llvm/lib/CodeGen/MachineSSAUpdater.cpp

```diff
@@ -236,6 +236,19 @@ void MachineSSAUpdater::RewriteUse(MachineOperand &U) {
     NewVR = GetValueInMiddleOfBlock(UseMI->getParent());
   }
 
+  // Insert a COPY if register class contraint isn't fulfilled.
+  if (const TargetRegisterClass *RC =
+          dyn_cast_or_null<const TargetRegisterClass *>(RegAttrs.RCOrRB)) {
+    if (NewVR && !RC->hasSubClassEq(MRI->getRegClass(NewVR))) {
```

Comment:
This should probably be using getCommonSubClass and checking if it's different from the original class

Reply:
What would the difference be compared to just using hasSubClassEq to check if the new class is subset of the original class?

Reply:
Classes don't always follow a strict subset relationship. This will try to find one that works

Reply:
Ok. Now when using `constrainRegClass ` instead I think this part should be fine then. Since that one is using `getCommonSubClass` internally.